src/Simplify.cpp

```diff
@@ -3205,6 +3233,18 @@ void simplify_test() {
     check((x - cast(Float(64), 0.5f)) * (x - cast(Float(64), 0.5f)),
           (x + Expr(-0.5)) * (x + Expr(-0.5)));
 
+    check(cast(Int(64).vector_of(3), ramp(x, 2, 3)),
+          ramp(cast(Int(64), x), cast(Int(64), 2), 3));
+
+    check(cast(Int(64), x + 1) - cast(Int(64), x), cast(Int(64), 1));
```

Comment:
Add a check where the outer type is Int(32) and the inner type is Int(64). That's the other case your new rule seems to imply.